File: scripts/list-linear.mjs
const fs = require('fs');
const https = require('https');

// Read LINEAR_API_KEY_B64 from .env.local, decode, use as API key
const envContent = fs.readFileSync('C:/Users/ADMIN/vietnamese-eden-mvp/.env.local', 'utf-8');
let apiKey = '';
for (const line of envContent.split(/\r?\n/)) {
  const m = line.match(/^LINEAR_API_KEY_B64=(.+)/);
  if (m) {
    apiKey = Buffer.from(m[1].trim().replace(/^"|"$/g, ''), 'base64').toString().trim();
    break;
  }
}

if (!apiKey) {
  console.error('LINEAR_API_KEY_B64 not found');
  process.exit(1);
}

const query = {
  query: `query {
    issues(
      filter: { 
        team: { key: { eq: "ALE" } }, 
        state: { type: { nin: ["completed", "canceled"] } } 
      }
      first: 10
      sortBy: priority
    ) {
      nodes {
        identifier
        title
        priority
        state { name }
      }
    }
  }`
};

const body = JSON.stringify(query);
const req = https.request('https://api.linear.app/graphql', {
  method: 'POST',
  headers: {
    'Authorization': apiKey,
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(body)
  }
}, res => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    const d = JSON.parse(data);
    if (d.errors) { console.error('GraphQL errors:', JSON.stringify(d.errors)); process.exit(1); }
    const issues = d.data.issues.nodes;
    console.log('Active issues:', issues.length);
    for (const i of issues) {
      const p = ['','Urgent','High','Medium','Low'][i.priority] || '';
      console.log(`  ${i.identifier} [${p}] ${i.state.name} — ${i.title.substring(0, 80)}`);
    }
  });
});
req.write(body);
req.end();

File: scripts/list-linear.py
import json, urllib.request, base64
from pathlib import Path

p = Path(r"C:\Users\ADMIN\vietnamese-eden-mvp\.env.local")
key_b64 = None
for line in p.read_text(encoding='utf-8').splitlines():
    s = line.strip()
    if s.startswith('LINEAR_API_KEY_B64=***        key_b64 = s.split('=', 1)[1].strip().strip('"')
        break
api_key = base64.b64decode(key_b64).decode().strip()

query = '''
query {
  issues(
    filter: { 
      team: { key: { eq: "ALE" } }, 
      state: { type: { nin: ["completed", "canceled"] } } 
    }
    first: 10
    sortBy: priority
  ) {
    nodes {
      identifier
      title
      priority
      state { name }
    }
  }
}
'''
req = urllib.request.Request(
    'https://api.linear.app/graphql',
    data=json.dumps({'query': query}).encode(),
    headers={'Authorization': api_key, 'Content-Type': 'application/json'}
)
resp = json.loads(urllib.request.urlopen(req, timeout=15).read().decode())
issues = resp['data']['issues']['nodes']
print(f'Active issues: {len(issues)}')
for i in issues:
    p_label = ['','Urgent','High','Medium','Low','No'][i['priority']]
    print(f'  {i["identifier"]} [{p_label}] {i["state"]["name"]} — {i["title"][:80]}')
